### PR TITLE
Backup Compatibility with Yours Wallet

### PR DESCRIPTION
**Pull request by petarvx**
This PR modifies the keys format in 1sat so they can be more compatible with Yours wallet when importing / exporting and storing.

### CODE REVIEW

**Comment by rohenaz** [comment]
the vercel deploy failed and has some errors i think might actually be valid.

**Comment by petarvx** [comment]
> the vercel deploy failed and has some errors i think might actually be valid.

You're right, sorry about that - fixed!

File: src/utils/wallet.ts [hunk 1]
```diff
@@ -7,3 +18,37 @@ export const getBalanceText = (balance: number, numDecimals: number) => {
 		? balance.toFixed(2)
 		: balance.toString();
 };
+
+export const backupKeys = () => {
+	let payPkDerivationPath = changeAddressPath.value;
+	let ordPkDerivationPath = ordAddressPath.value;
+
+	if (typeof changeAddressPath.value === "number") {
+		payPkDerivationPath = `m/0/0/0/0/${changeAddressPath.value}`;
```
Comment (rohenaz) [inline src/utils/wallet.ts]: this should be `m/${changeAddressPath.value}`

File: src/utils/wallet.ts [hunk 2]
```diff
@@ -7,3 +18,37 @@ export const getBalanceText = (balance: number, numDecimals: number) => {
...
+	}
+
+	if (typeof ordAddressPath.value === "number") {
+		ordPkDerivationPath = `m/0/0/0/0/${ordAddressPath.value}`;
```
Comment (rohenaz) [inline src/utils/wallet.ts]: this should be `m/${ordAddressPath.value}`

**Review by rohenaz — CHANGES_REQUESTED** [review]
I've tested this a bit and found a problem with derivation paths in the backup file after creating a new wallet.

- create a new wallet, export the keys
You will see derivation paths like
"payDerivationPath": "m/0/0/0/0/0"
but the key produced actually corresponds to "m/0" as it should

How I verified this:
- set aside the payment address for the new wallet to be verified
- go to bip39 tool: https://iancoleman.io/bip39/
- paste in the mnemonic from the backup file
- type in the base path "m"
- look at the produced addresses, you'll see m/0 (first address) corresponds to the payment address

should be an easy fix we just need to update the derivation paths to be m/0 for payment, m/X where X is the first 1s address. I've noted the lines that need to be changed.

same for ord key i got m/0/0/0/0/151 in the backup file instead of m/151

also we need to backport from alpha again there are some fixes in there

**Comment by petarvx** [comment]
> I've tested this a bit and found a problem with derivation paths in the backup file after creating a new wallet.
> 
> * create a new wallet, export the keys
>   You will see derivation paths like
>   "payDerivationPath": "m/0/0/0/0/0"
>   but the key produced actually corresponds to "m/0" as it should
> 
> How I verified this:
> 
> * set aside the payment address for the new wallet to be verified
> * go to bip39 tool: https://iancoleman.io/bip39/
> * paste in the mnemonic from the backup file
> * type in the base path "m"
> * look at the produced addresses, you'll see m/0 (first address) corresponds to the payment address
> 
> should be an easy fix we just need to update the derivation paths to be m/0 for payment, m/X where X is the first 1s address. I've noted the lines that need to be changed.
> 
> same for ord key i got m/0/0/0/0/151 in the backup file instead of m/151
> 
> also we need to backport from alpha again there are some fixes in there

Thank you for the explanation, that helped a lot! Updated 👍

**Comment by petarvx** [comment]
Pushed an update to not assign payPk as identityPk as default when there is no identity key. Instead, we just won't have identity key if there isn't any.

**Review by rohenaz — APPROVED** [review]

**Pull request merged**